src/components/ha-icon-picker.ts

```diff
@@ -87,46 +131,37 @@ export class HaIconPicker extends LitElement {
     `;
   }
 
-  private async _openedChanged(ev: PolymerChangedEvent<boolean>) {
-    this._opened = ev.detail.value;
-    if (this._opened && !iconLoaded) {
-      const iconList = await import("../../build/mdi/iconList.json");
-
-      iconItems = iconList.default.map((icon) => ({
-        icon: `mdi:${icon.name}`,
-        keywords: icon.keywords,
-      }));
-      iconLoaded = true;
-
-      this.comboBox.filteredItems = iconItems;
-
-      Object.keys(customIcons).forEach((iconSet) => {
-        this._loadCustomIconItems(iconSet);
-      });
-    }
-  }
-
-  private async _loadCustomIconItems(iconsetPrefix: string) {
-    try {
-      const getIconList = customIcons[iconsetPrefix].getIconList;
-      if (typeof getIconList !== "function") {
-        return;
+  private _filterIcons = memoizeOne(
+    (filterString: string, iconItems = ICONS) => {
+      if (!filterString) {
+        return iconItems;
+      }
+      const startTime = performance.now();
+      const filteredItems = iconItems.filter(
+        (item) =>
+          item.icon.includes(filterString) ||
+          item.keywords.some((word) => word.includes(filterString))
+      );
+      if (__DEV__) {
```

Comment:
It's better to remove this. We do not want to overloaded the console in dev.

Reply:
Yeah I can get rid of it, but it might be useful if we start messing with a better rank and sort.